src/PostgREST/Query/QueryBuilder.hs

```diff
@@ -111,12 +111,12 @@ mutateRequestToQuery (Update mainQi uCols body logicForest pkFlts range ordts re
     SQL.sql $ "SELECT " <> emptyBodyReturnedColumns <> " FROM " <> fromQi mainQi <> " WHERE false"
 
   | range == allRange =
-    let whereLogic | null logicForest = if null pkFlts then "FALSE" else pgrstUpdateBodyF
-                   | otherwise        = logicForestF in
+    let whereLogic | null logicForest = if null pkFlts || not colsHavePk then mempty else " WHERE " <> pgrstUpdateBodyF
```

Comment:
I'm puzzled by this. By looking at this block, seems like a `WHERE` is applied all the time. Can it be clarified?